share/functions/alias.fish

```diff
@@ -58,7 +61,32 @@ function alias --description 'Creates a function wrapping a command'
     end
     set -l cmd_string (string escape -- "alias $argv")
     set wrapped_cmd (string join ' ' -- $first_word $body | string escape)
-    echo "function $name --wraps $wrapped_cmd --description $cmd_string; $prefix $first_word $body \$argv; end" | source
+    if set -q _flag_expand_next || test $has_trailing_space -eq 0
+        echo "function $name --wraps $wrapped_cmd --description $cmd_string
+            if set -q \$argv
+                $prefix $first_word $body
+                return
+            end
+
+            set -l tmp (string split -m 1 ' ' -- \$argv)
+            set -l inner_name \$tmp[1]
+            set -l inner_body \$tmp[2]
+
+            for func in (alias)
```

Comment:
Since aliases are just simple functions, I wonder if makes sense to run whatever `functions $inner_name` prints to include functions, too. Probably not worth it.